com.unity.ml-agents/Runtime/Actuators/ActuatorDiscreteActionMask.cs

```diff
@@ -27,11 +27,12 @@ internal class ActuatorDiscreteActionMask : IDiscreteActionMask
         /// </summary>
         public int CurrentBranchOffset { get; set; }
 
-        internal ActuatorDiscreteActionMask(IList<IActuator> actuators, int sumOfDiscreteBranchSizes, int numBranches)
+        internal ActuatorDiscreteActionMask(IList<IActuator> actuators, int sumOfDiscreteBranchSizes, int numBranches, int[] branchSizes = null)
```

Comment:
I can make branchSizes required; that would mean changing some tests, but we can remove code from LazyInitialize.

Reply:
i wanted to avoid the allocation if the mask is never used.

Reply:
Ah, I see. I should at least be able to change CombineActionSpecs() to use Array.Empy if there are no discrete actions, but I think we always need to do the aggregation